ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/macro/MacroRegistryImpl.java

```diff
@@ -39,9 +38,7 @@ public MacroRegistryImpl() {
     public void register(Set<Macro> macros) {
         for (Macro macro : macros) {
             final String name = macro.getName();
-            if (this.macros.containsKey(name)) {
-                Log.warn(MacroRegistryImpl.class, "Command macro '" + name + "' is already registered.");
```

Comment:
> too much outputs in browser console

Is it a real reason why you've removed that code? :smirk_cat: 
If it is then the main reason of that warning must be fixed but removing the logging is the bad way to fix it, of course.

Ok, let's think about that all together :family_man_boy_boy: because we're using such approach in several places and I think it's really important to discuss it.

Imagine the following situation: some developer works on his own extension and wants to register his implementation of Macro. But another Macro with the same id/name is already registered in IDE. How we should inform the plugin's developer (not IDE user!) that he should choose another unique id/name for registering a Macro? Or do you think we shouldn't at all?

@vitaliy-guliy @vparfonov @evidolob @dkuleshov I'm interested in your opinion?

IMO informing **plugin's developer** about such kind of situations in browser's DevConsole is the best way in our case. And I think it's one the main purpose of the browser's DevConsole - outputting really useful messages for the developer in order to help him to solve his problems in the shortest period of time.